src/common_info.rs

```diff
@@ -0,0 +1,57 @@
+use std::io::{self, Read};
+
+use serde::Serialize;
+
+use crate::read::ReadExt;
+
+#[derive(Debug, Serialize)]
+pub struct CommonInfo {
+    pub note_id: u64,
+    pub reply_to_note_id: u64,
+    pub sender_pid: u32,
+    pub sent: BasicDateTime,
+}
+
+#[derive(Debug, Serialize)]
+pub struct BasicDateTime {
+    pub year: u8,
+    pub month: u8,
+    pub day: u8,
+    pub hour: u8,
+    pub minute: u8,
+    pub second: u8,
+}
+
+impl TryFrom<&[u8]> for CommonInfo {
+    type Error = io::Error;
+
+    fn try_from(mut value: &[u8]) -> Result<Self, Self::Error> {
+        value.take(0x18);
```

Comment:
This probably doesn't work, try this instead
```suggestion
    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
        let mut value = &value[0x18..];
```